bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -0,0 +1,656 @@
+
+/**
+ * The {@link RadioThermostatHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * Based on the 'airquality' binding by Kuba Wolanin
+ *
+ * @author Michael Lobstein - Initial contribution
+ */
+public class RadioThermostatHandler extends BaseThingHandler {
+    private final RadioThermostatStateDescriptionProvider stateDescriptionProvider;
+
+    private Logger logger = LoggerFactory.getLogger(RadioThermostatHandler.class);
+
+    private static final String URL = "http://%hostName%/%resource%";
+    
+    private static final String DEFAULT_RESOURCE    = "tstat";
+    private static final String RUNTIME_RESOURCE    = "tstat/datalog";
+    private static final String HUMIDITY_RESOURCE   = "tstat/humidity";
+    private static final String MODEL_RESOURCE      = "tstat/model";
+    private static final String NAME_RESOURCE       = "sys/name";
+
+    private static final int DEFAULT_REFRESH_PERIOD = 2;
+    private static final int DEFAULT_LOG_REFRESH_PERIOD = 10;
+
+    private ScheduledFuture<?> refreshJob;
+    private ScheduledFuture<?> logRefreshJob;
+
+    private RadioThermostatData rthermData = new RadioThermostatData();
+    
+    private Gson gson;
+
+    private int initialized = 0;
+    private int retryCounter = 0;
+
+    public RadioThermostatHandler(Thing thing, RadioThermostatStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        gson = new Gson();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing RadioThermostat handler.");
+
+        RadioThermostatConfiguration config = getConfigAs(RadioThermostatConfiguration.class);
+        logger.debug("config hostName = {}", config.hostName); 
+        logger.debug("config refresh = {}", config.refresh);
+        logger.debug("config logRefresh = {}", config.logRefresh);
+        logger.debug("config disableLogs = {}", config.disableLogs);
+        logger.debug("config disableHumidity = {}", config.disableHumidity);
+
+        String errorMsg = null;
+
+        if (StringUtils.trimToNull(config.hostName) == null) {
+            errorMsg = "Parameter 'hostName' is mandatory and must be configured";
+        }
+        if (config.refresh != null && config.refresh < 1) {
+            errorMsg = "Parameter 'refresh' must be at least 1 minute";
+        }
+        if (config.logRefresh != null && config.logRefresh < 5) {
+            errorMsg = "Parameter 'logRefresh' must be at least 5 minutes";
+        }
+        if (config.disableLogs != null && (config.disableLogs != 0 && config.disableLogs != 1)) {
+            errorMsg = "Parameter 'disableLogs' must be either 0 or 1";
+        }
+        if (config.disableHumidity != null && (config.disableHumidity != 0 && config.disableHumidity != 1)) {
+            errorMsg = "Parameter 'disableHumidity' must be either 0 or 1";
+        }
```

Comment:
You don't need these checks, since you already specified min/max in the XML file.